scp.py

```diff
@@ -167,7 +167,7 @@ def put(self, files, remote_path=b'.',
 
         self.close()
 
-    def putfo(self, fl, remote_path, mode='0644'):
+    def putfo(self, fl, remote_path, mode='0644', length=None):
```

Comment:
``size`` might be a better parameter name for this, just to match the parameter names of ``_send_file``.